### PR TITLE
Export moderated posts

### PR DESCRIPTION
**Pull request by tdesvenain**
DEVAS-1963

### CODE REVIEW

File: assembl/locale/assembl-v2/de.po [hunk 1]
```diff
@@ -1979,6 +2011,30 @@ msgstr "Textwert"
 msgid "administration.thematic"
 msgstr "Thema"
 
+#. [administration.theme.firstColor]
+#. defaultMessage is:
+#. Primary color
+#: messages.json
+#, fuzzy
```
Comment (ayyazdaniaryan) [inline assembl/locale/assembl-v2/de.po]: Resolve these fuzzies by hand, please :)

File: assembl/locale/assembl-v2/tr.po [hunk 2]
```diff
@@ -3947,6 +4035,14 @@ msgstr "Bu gönderiye hiç cevap verilmemiş"
 msgid "debate.thread.numberOfResponses_1"
 msgstr "%{count} cevap"
 
+#. [debate.thread.overviewPosts]
+#. defaultMessage is:
+#. Overview
+#: messages.json
+#, fuzzy
+msgid "debate.thread.overviewPosts"
+msgstr "Daha fazla görüş"
```
Comment (ayyazdaniaryan) [inline assembl/locale/assembl-v2/tr.po]: Are these google translated?

File: assembl/models/auth.py [hunk 3]
```diff
@@ -1014,7 +1014,7 @@ def validate_password(self, password):
                     [results['feedback']['warning']] + results['feedback']['suggestions'])
         # refuse if reusing an old password
         for p in self.old_passwords:
-            if verify_password(password, p.password if p else ""):
+            if verify_password(password, p.password if p.password else ""):
```
Comment (ayyazdaniaryan) [inline assembl/models/auth.py]: Hehe, this was resolved on develop as well.

File: assembl/tests/views_tests/test_discussion_export.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from assembl.views.api2.discussion import thread_csv_export
+from assembl.views.traversal import InstanceContext
+from freezegun import freeze_time
+
+
+@freeze_time("2018-3-8")
+def test_thread_csv_export(request, discussion, thread_phase_root_idea, top_post_in_thread_phase, second_post_in_thread_phase, third_post_in_thread_phase):
+    request.GET = {}
+    assert thread_phase_root_idea.message_view_override in ('thread', None)
+    import pdb; pdb.set_trace()
```
Comment (ayyazdaniaryan) [inline assembl/tests/views_tests/test_discussion_export.py]: Remove this pdb please

File: assembl/tests/views_tests/test_discussion_export.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+from assembl.views.api2.discussion import thread_csv_export
+from assembl.views.traversal import InstanceContext
+from freezegun import freeze_time
+
+
+@freeze_time("2018-3-8")
+def test_thread_csv_export(request, discussion, thread_phase_root_idea, top_post_in_thread_phase, second_post_in_thread_phase, third_post_in_thread_phase):
+    request.GET = {}
+    assert thread_phase_root_idea.message_view_override in ('thread', None)
```
Comment (ayyazdaniaryan) [inline assembl/tests/views_tests/test_discussion_export.py]: Shouldn't there always be a root idea (which by default makes it a thread phase)?

File: assembl/utils.py [hunk 6]
```diff
@@ -116,6 +119,7 @@ def get_vote_session_ideas(discussion, start=None, end=None):
 def get_ideas_for_export(discussion, module_type=None, start=None, end=None):
     ideas = []
     for phase in discussion.timeline_events:
+        import pdb; pdb.set_trace()
```
Comment (ayyazdaniaryan) [inline assembl/utils.py]: Please remove this pdb as well

File: assembl/views/api2/discussion.py [hunk 7]
```diff
@@ -810,16 +841,20 @@ def users_csv_export(request):
                     users[user.id][key] += ', ' + value
 
     for post in posts:
+        if not post.id:
+            continue
```
Comment (ayyazdaniaryan) [inline assembl/views/api2/discussion.py]: Huh? In what scenario will there be posts in the db that do not have an id?

File: assembl/views/api2/discussion.py [hunk 8]
```diff
@@ -2325,21 +2369,23 @@ def thread_csv_export(request):
         i = fieldnames.index(SENTIMENT_ACTOR_EMAIL.encode('utf-8')) + 1
         fieldnames[i:i] = ['sentiment ' + name.encode('utf-8') for (name, path) in extra_columns_info]
 
+    publication_states = get_publication_states(request) or [PublicationStates.PUBLISHED]
+    import pdb; pdb.set_trace()
```
Comment (ayyazdaniaryan) [inline assembl/views/api2/discussion.py]: Please remove this pdb as well

File: assembl/utils.py [hunk 9]
```diff
@@ -13,15 +13,19 @@ def format_date(datetime_to_format):
     return datetime_to_format.strftime('%d/%m/%Y')
 
 
-def get_posts(idea, start=None, end=None, include_deleted=None):
+def get_posts(idea, start=None, end=None,
+              include_deleted=None,
+              publication_states=None,
```
Comment (ayyazdaniaryan) [inline assembl/utils.py]: So, given that in a lot of the export funcs you pass a default `PublicationStates.published`, perhaps you should make this state the default here. Also, update the docstrings below to reflect the new parameters that you've added.

File: assembl/locale/assembl-v2/de.po (hunk 1 above)
Comment (tdesvenain) [reply assembl/locale/assembl-v2/de.po]: It's not my change. I think this pr is messy, i will recreate it from scratch

File: assembl/locale/assembl-v2/tr.po (hunk 2 above)
Comment (tdesvenain) [reply assembl/locale/assembl-v2/tr.po]: not my change... cf previous

File: assembl/utils.py (hunk 6 above)
Comment (tdesvenain) [reply assembl/utils.py]: the last commit is WIP, sorry...

File: assembl/views/api2/discussion.py (hunk 7 above)
Comment (tdesvenain) [reply assembl/views/api2/discussion.py]: this happens in test.. dnw

File: assembl/utils.py (hunk 9 above)
Comment (tdesvenain) [reply assembl/utils.py]: as None value is available, i don't like having a non None default value
i fix the doc

**Pull request closed without merging**